janusgraph-inmemory/src/main/java/org/janusgraph/diskstorage/inmemory/BufferPage.java

```diff
@@ -111,24 +113,64 @@ private int getEntryEndOffset(final int index) {
         }
     }
 
+    static int readValPos(byte[] rawData, int entryOffset, ByteBuffer entryBuffer)
```

Comment:
It took me some time to understand this method thoroughly. I think the usage of entryBuffer here is not straight-forward to understand, since you are kind of mixing the two use cases (reading and moving cursor v.s. purely reading) together in one method. Would be helpful if you can add some javadoc.

Reply:
done